File: pe-core/src/main/java/org/saar/core/common/flatreflected/FlatReflectedMeshWriter.java
package org.saar.core.common.flatreflected;

import org.saar.core.mesh.writers.ElementsMeshWriter;

public class FlatReflectedMeshWriter implements ElementsMeshWriter<FlatReflectedVertex> {

    private final FlatReflectedMeshPrototype prototype;

    public FlatReflectedMeshWriter(FlatReflectedMeshPrototype prototype) {
        this.prototype = prototype;
    }

    @Override
    public void writeVertex(FlatReflectedVertex vertex) {
        this.prototype.getPositionBuffer().getWriter().write3f(vertex.getPosition3f());
    }

    @Override
    public void writeIndex(int index) {
        this.prototype.getIndexBuffer().getWriter().writeInt(index);
    }
}

File: pe-core/src/main/kotlin/org/saar/core/common/flatreflected/FlatReflectedMeshWriter.kt
package org.saar.core.common.flatreflected

import org.saar.core.mesh.writer.IndexedMeshWriter
import org.saar.core.mesh.writer.VertexMeshWriter
import org.saar.lwjgl.util.DataWriter

class FlatReflectedMeshWriter(
    private val positionWriter: DataWriter,
    private val indexWriter: DataWriter,
) : VertexMeshWriter<FlatReflectedVertex>, IndexedMeshWriter {

    override fun writeVertex(vertex: FlatReflectedVertex) {
        this.positionWriter.write3f(vertex.position3f)
    }

    override fun writeIndex(index: Int) = this.indexWriter.writeInt(index)
}